src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/listeners/HMPPSAuditListener.kt

```diff
@@ -33,7 +33,7 @@ class HMPPSAuditListener(
     @Id
     @GeneratedValue
     @Hidden
-    val id: UUID? = null,
+    var id: UUID? = null,
```

Comment:
Just curious why this is `var` vs `val`?

Reply:
we need the ID to be mutable because the audit event comes in with a null ID, then has its ID set by the auditS3Service if it's getting saved to an S3 bucket. Previously we didn't need to do this as the ID would be set automatically in the table once it was saved to the database